### DISCUSSION TITLE
surface density projection?

### DISCUSSION THREAD

**Discussion by BenWibking** [Alpine-DAV/ascent#951] [Q&A] [answered]
Is there an extract that can be used to do a 2D projection of a field, say, projecting the volumetric density field to a surface density, similar to e.g., yt's ProjectionPlot: https://yt-project.org/doc/visualizing/plots.html#types-of-projections?

**Comment by cyrush** [comment, marked as answer]
@BenWibking sorry for the delay with answering. 

We don't have anything that does an integrated projection of rays.

We have volume rendering and an x-ray extract (which takes into consideration emission and absorption provided as mesh fields), but we don't have something that does straight forward integration or general field weighted integration along rays. 

For similar use cases - we used Binning as a coarse way to collapse:

https://ascent.readthedocs.io/en/latest/Actions/Binning.html

If you are axis aligned, a 2D spatial binning with a sum collapse can give a volume or mass projection. 

Our Binning Tutorial shows how to set these up:
https://ascent.readthedocs.io/en/latest/Tutorial_Intro_Binning.html

An integration option would be very similar to our x-ray extracts, we could look into adding in the future.

**Reply by BenWibking** [reply]
Axis aligned is all I need for now. Thanks.
